apis/python/src/tiledbsoma/_collection.py

```diff
@@ -509,9 +509,49 @@ class Collection(
     CollectionBase[CollectionElementType], somacore.Collection[CollectionElementType]
 ):
     """
-    A persistent collection of SOMA objects, mapping string keys to any SOMA object.
+    ``Collection`` is a persistent container of named SOMA objects, stored as
+    a mapping of string keys and SOMA object values. Values may be any
+    persistent ``tiledbsoma`` object, including ``DataFrame``,
+    ``SparseNDArray``, ``DenseNDArray``, ``Experiment``, ``Measurement``,
+    or another ``Collection``. A ``Collection`` refers to elements by a
+    per-element URI. A ``Collection`` may store its reference to an
+    element by absolute URI or relative URI.
 
     [lifecycle: experimental]
+
+    Examples:
+    ---------
+    >>> import tiledbsoma
+    >>> import pyarrow as pa
+    >>> import numpy as np
+    >>> # create a collection and add a (10,10) dense matrix to it
+    ... with tiledbsoma.Collection.create("./test_collection") as my_collection:
+    ...     with my_collection.add_new_dense_ndarray(
```

Comment:
These nested `with`s are no longer needed because collections now keep track of their children, so this can be:

```python
with create(...) as my_collection:
  my_dense_ndarray = my_collection.add_new_dense_ndarray(...)
  # do stuff with my_dense_ndarray
```